game-core/src/main/java/games/strategy/triplea/delegate/battle/MustFightBattle.java

```diff
@@ -1086,28 +1088,38 @@ private void pushFightLoopOnStack(final boolean firstRun) {
   private void addFightStartSteps(final boolean firstRun, final List<IExecutable> steps) {
     final boolean offensiveAa = canFireOffensiveAa();
     final boolean defendingAa = canFireDefendingAa();
-    if (offensiveAa) {
-      steps.add(
-          new FireOffensiveAaGuns() {
-            private static final long serialVersionUID = 3802352588499530533L;
-
-            @Override
-            public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
-              fireOffensiveAaGuns();
-            }
-          });
+    final BattleStep offensiveAaStep = new OffensiveAaFire(this, this);
+    final BattleStep defensiveAaStep = new DefensiveAaFire(this, this);
+    if (offensiveAaStep.valid()) {
+      steps.add(offensiveAaStep);
     }
-    if (defendingAa) {
-      steps.add(
-          new FireDefensiveAaGuns() {
-            private static final long serialVersionUID = -1370090785540214199L;
+    // see Save Game Compatibility Note on getBattleExecutables
+    new IExecutable() {
+      private static final long serialVersionUID = 3802352588499530533L;
 
-            @Override
-            public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
-              fireDefensiveAaGuns();
-            }
-          });
+      @Override
+      public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
+        // if this gets deserialized, then forward the work to the new BattleStep
+        final BattleStep offensiveAaStep =
+            new OffensiveAaFire(MustFightBattle.this, MustFightBattle.this);
+        offensiveAaStep.execute(stack, bridge);
+      }
+    };
+    if (defensiveAaStep.valid()) {
+      steps.add(defensiveAaStep);
     }
+    // see Save Game Compatibility Note on getBattleExecutables
```

Comment:
Ditto, we may not need this comment. Someone modifying a method should be aware of the method contract, which implies they would have at least read the method javadoc. Removing this comment could reduce future maintenance burden (less stuff to keep in sync, code and comments need to always be verified and kept in sync).

Reply:
My only fear is that these old steps are inside of methods that are called by `getBattleExecutables`.  So if you go to the top of just this method, you won't see the note about save compatibility.  But I'll remove the note.